core/src/main/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbe.java

```diff
@@ -0,0 +1,112 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.LogCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import java.time.ZonedDateTime;
+import java.util.regex.Matcher;
+
+/**
+ * Using the analysis done by {@link SCMLinkValidationProbe},
+ * this probe will find number of unreleased commits in a repository
+ */
+@Component
+@Order(value = HasUnreleasedProductionChangesProbe.ORDER)
+public class HasUnreleasedProductionChangesProbe  extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HasUnreleasedProductionChangesProbe.class);
+
+    public static final int ORDER = LastCommitDateProbe.ORDER + 100;
+    public static final String KEY = "unreleased-production-changes";
+
+    @Override
+    public ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        final Matcher matcher = SCMLinkValidationProbe.GH_PATTERN.matcher(plugin.getScm());
+        if (!matcher.find()) {
+            return ProbeResult.failure(key(), "The SCM link is not valid");
+        }
+        final String repo = String.format("https://%s/%s", matcher.group("server"), matcher.group("repo"));
+        final String folder = matcher.group("folder");
+
+        try (Git git = Git.cloneRepository().setURI(repo).setDirectory(context.getScmRepository().toFile()).call()) {
+            final LogCommand logCommand = git.log().setMaxCount(1);
+            if (folder != null) {
+                logCommand.addPath(folder);
+            }
+            logCommand.addPath("pom.xml");
+            logCommand.addPath("src/");
+
+            final RevCommit commit = logCommand.call().iterator().next();
+            if (commit == null) {
+                return ProbeResult.failure(key(), "Last commit cannot be extracted. Please validate sub-folder if any.");
+            }
+            final ZonedDateTime commitDate = ZonedDateTime.ofInstant(
+                commit.getAuthorIdent().getWhenAsInstant(),
+                commit.getAuthorIdent().getZoneId()
+            );
+            context.setLastCommitDate(commitDate);
+            return ProbeResult.success(key(), commitDate.toString());
```

Comment:
what if the `commit` object is not null and has a date that is more recent than the plugin release timestamp?

Reply:
> what if the `commit` object is not null and has a date that is more recent than the plugin release timestamp?

Then the test should fail.

Reply:
yes, but would the current code fail it by returning `ProbeResult.failure()`?
I'm asking because I'm a little confused with the code present in the `HasUnreleasedProductionChangesProbe.java` file.

Reply:
> yes, but would the current code fail it by returning ProbeResult.failure()?

The correct code should fail it. If it does not then I need to modify the code.

Reply:
Then you need to modify the code.